### PR TITLE
Add max-services and max-namespaces workloads

### PR DESCRIPTION
**Pull request by rsevilla87**
Also adding instant Prometheus queries for kube_node_role and update kubelet metrics as topk doesn't get along well with rate.

Update: pin_server now accepts generic nodeSelector

Signed-off-by: Raul Sevilla <rsevilla@redhat.com>

### CODE REVIEW

File: docs/kube-burner.md [hunk 1]
```diff
@@ -36,6 +36,15 @@ Each iteration of this workload creates the following objects:
   - 1 deployment holding a client application for the previous database
   - 1 service pointing to the postgresl database
```
Comment (smalleni) [inline docs/kube-burner.md:38]: space between as and many :)

File: docs/kube-burner.md [hunk 2]
```diff
@@ -36,6 +36,15 @@ Each iteration of this workload creates the following objects:
...
 
+- **max-namespaces**: This workload is a cluster limits focused test which creates maximum possible namespaces across the cluster. This is a namespaced workload, meaning that kube-burner **will createas many namespaces with these objects as the configured job_iterations**. Each namespace will be populated with the following objects.:
+  - 1 deployment holding a postgresql database
+  - 1 deployment holding a client application for the previous database
+  - 1 service pointing to the postgresl database
+
+- **max-services**: This workload is a cluster limits focused test which creates maximum possible services per each namespace. Each namespace will be populated with the following objects.:
```
Comment (smalleni) [inline docs/kube-burner.md:44]: Maybe add that job_iterations defines the number of namespaces?

Comment (rsevilla87) [reply docs/kube-burner.md]: no, this workload only creates one namespace, I've made some modifications in docs to clarify this.

Comment (smalleni) [reply docs/kube-burner.md]: Ack, thanks!

File: docs/kube-burner.md [hunk 3]
```diff
@@ -36,6 +36,15 @@ Each iteration of this workload creates the following objects:
   - 1 deployment holding a client application for the previous database
   - 1 service pointing to the postgresl database
 
+- **max-namespaces**: This workload is a cluster limits focused test which creates maximum possible namespaces across the cluster. This is a namespaced workload, meaning that kube-burner **will create as many namespaces with these objects as the configured job_iterations**.
+  - 1 deployment holding a postgresql database
```
Comment (chaitanyaenr) [inline docs/kube-burner.md:40]: We might want to make the max-namespaces test a little bit more heavy. We usually have 10k namespaces with ~60k pods using 10k deployments, ~190k secrets, image streams, environment variables and services - https://github.com/openshift/svt/blob/master/openshift_scalability/config/golang/cluster-limits-namespaces.yaml. Thoughts?

Comment (rsevilla87) [reply docs/kube-burner.md:40]: makes sense, I didn't realize that deployment had 6 replicas

File: docs/kube-burner.md [hunk 4]
```diff
@@ -36,6 +36,15 @@ Each iteration of this workload creates the following objects:
...
+  - 5 deployments consisting of a client application for the previous database
+  - 1 service pointing to the postgresl database
+
+- **max-services**: This workload is a cluster limits focused test which creates maximum possible services per namespace. It **will create a single namespace, each iteration of this workload will populate that namespace with these objects:**
```
Comment (chaitanyaenr) [inline docs/kube-burner.md:44]: The number of pods/deployments that can run with 5000 services in the namespace is limited by the byte size of the service name as there's a limit on the size of the arguments that be passed during exec. Should we expose the service name as a config option to be able to tweak it during the test? Thoughts?

File: docs/kube-burner.md (hunk 3 above)
Comment (rsevilla87) [reply docs/kube-burner.md:40]: fixed!

File: docs/kube-burner.md (hunk 4 above)
Comment (rsevilla87) [reply docs/kube-burner.md:44]: As we discussed offline, I removed UUID suffix from resource names. so we shouldn't overcome the byte size limit.

**Review by chaitanyaenr — APPROVED** [review]
LGTM, nice work @rsevilla87.

**Pull request merged**